jetpack.php

```diff
@@ -12,94 +12,38 @@
 define( 'JETPACK__PLUGIN_FILE',        __FILE__ );
-
-defined( 'JETPACK_CLIENT__AUTH_LOCATION' )   or define( 'JETPACK_CLIENT__AUTH_LOCATION', 'header' );
-defined( 'JETPACK_CLIENT__HTTPS' )           or define( 'JETPACK_CLIENT__HTTPS', 'AUTO' );
 defined( 'JETPACK__GLOTPRESS_LOCALES_PATH' ) or define( 'JETPACK__GLOTPRESS_LOCALES_PATH', JETPACK__PLUGIN_DIR . 'locales.php' );
-defined( 'JETPACK__API_BASE' )               or define( 'JETPACK__API_BASE', 'https://jetpack.wordpress.com/jetpack.' );
-defined( 'JETPACK_PROTECT__API_HOST' )       or define( 'JETPACK_PROTECT__API_HOST', 'https://api.bruteprotect.com/' );
-defined( 'JETPACK__WPCOM_JSON_API_HOST' )    or define( 'JETPACK__WPCOM_JSON_API_HOST', 'public-api.wordpress.com' );
-
 defined( 'JETPACK__SANDBOX_DOMAIN' ) or define( 'JETPACK__SANDBOX_DOMAIN', '' );
 
-defined( 'JETPACK__DEBUGGER_PUBLIC_KEY' ) or define(
-	'JETPACK__DEBUGGER_PUBLIC_KEY',
-	"\r\n" . '-----BEGIN PUBLIC KEY-----' . "\r\n"
-	. 'MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAm+uLLVoxGCY71LS6KFc6' . "\r\n"
-	. '1UnF6QGBAsi5XF8ty9kR3/voqfOkpW+gRerM2Kyjy6DPCOmzhZj7BFGtxSV2ZoMX' . "\r\n"
-	. '9ZwWxzXhl/Q/6k8jg8BoY1QL6L2K76icXJu80b+RDIqvOfJruaAeBg1Q9NyeYqLY' . "\r\n"
-	. 'lEVzN2vIwcFYl+MrP/g6Bc2co7Jcbli+tpNIxg4Z+Hnhbs7OJ3STQLmEryLpAxQO' . "\r\n"
-	. 'q8cbhQkMx+FyQhxzSwtXYI/ClCUmTnzcKk7SgGvEjoKGAmngILiVuEJ4bm7Q1yok' . "\r\n"
-	. 'xl9+wcfW6JAituNhml9dlHCWnn9D3+j8pxStHihKy2gVMwiFRjLEeD8K/7JVGkb/' . "\r\n"
-	. 'EwIDAQAB' . "\r\n"
-	. '-----END PUBLIC KEY-----' . "\r\n"
-);
-
-/**
- * Returns the location of Jetpack's lib directory. This filter is applied
- * in require_lib().
- *
- * @since 4.0.2
- *
- * @return string Location of Jetpack library directory.
- *
- * @filter require_lib_dir
- */
-function jetpack_require_lib_dir() {
-	return JETPACK__PLUGIN_DIR . '_inc/lib';
+// ensure a default autoload mechanism is available
+// this is filterable so on future platforms we can use a single autoloader
+// that compiles all plugin dependencies in one place
+if ( ! has_filter( 'jetpack_autoload' ) ) {
+	add_filter( 'jetpack_autoload', function( $plugin, $loader ) {
+		return require_once plugin_dir_path( $plugin ) . '/vendor/autoload.php';
+	}, 10, 2 );
 }
 
+$loader = apply_filters( 'jetpack_autoload', __FILE__, false );
 
-/**
- * Checks if the code debug mode turned on, and returns false if it is. When Jetpack is in
- * code debug mode, it shouldn't use minified assets. Note that this filter is not being used
- * in every place where assets are enqueued. The filter is added at priority 9 to be overridden
- * by any default priority filter that runs after it.
- *
- * @since 6.2.0
- *
- * @return boolean
- *
- * @filter jetpack_should_use_minified_assets
- */
-function jetpack_should_use_minified_assets() {
-	if ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ) {
-		return false;
-	}
-	return true;
-}
+// this should autoload the bootstrap file
+$plugin = new \Jetpack\V7\Core\Bootstrap();
```

Comment:
I was thinking having the major version in the class would help when it comes time to make breaking changes, since we could (temporarily) allow v8 to work alongside v7 without class name conflicts.

Reply:
I agree that this is a valid reason, but we have been dealing with version inconsistencies in Jetpack for a while now, and we have been able to use code logic to avoid any problems somewhat successfully. Or do you foresee a case when we will have to use both versions in the same Jetpack build?

Reply:
I tend to agree with @zinigor here - FWIW loading multiple versions of a package can lead to a lot of weird problems - hooking multiple instances of the same filters and actions, or doing the same preparatory/cleanup actions multiple times could lead to unexpected results (as pointed out by @lezama in https://github.com/Automattic/jpop-issues/issues/3882#issuecomment-492471936 too). The Jetpack codebase isn't prepared to handle that, and we'd need a bunch of additional work and tests to make sure we're supporting it correctly. 

I also think that we're used to keeping backwards compatibility because that's the WordPress way, and that's not a bad thing. Theoretiacally, loading a breaking v8 together with v7 would be against that, so I'd prefer avoiding it in favor of a better effort to keep things consistent and ship breaking changes together in the same release.